src/WebJobs.Script/Description/Binding/BindingMetadata.cs

```diff
@@ -110,30 +101,7 @@ public static BindingMetadata Create(JObject raw, INameResolver nameResolver = n
             bindingMetadata.Type = bindingType;
             bindingMetadata.Direction = bindingDirection;
             bindingMetadata.Connection = connection;
-
-            if (nameResolver != null)
-            {
-                nameResolver.ResolveAllProperties(bindingMetadata);
-
-                // We want to pass resolved metadata values into
-                // binding extensions, so we resolve fully here
```

Comment:
This was the code causing the issue. It was added in the last release. We were early binding all properties, including %rand-guid%.

We now allow the various extensions to resolve their properties themselves.